eth/downloader/queue.go

```diff
@@ -23,25 +23,32 @@ import (
 	"errors"
 	"fmt"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethdb"
 	"github.com/ethereum/go-ethereum/logger"
 	"github.com/ethereum/go-ethereum/logger/glog"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/rcrowley/go-metrics"
 	"gopkg.in/karalabe/cookiejar.v2/collections/prque"
 )
 
 var (
-	blockCacheLimit = 8 * MaxBlockFetch // Maximum number of blocks to cache before throttling the download
+	blockCacheLimit = 1024 // Maximum number of blocks to cache before throttling the download
```

Comment:
*[Well, currently there's no notion of a full node, only archive and light. This PR started out as a special sync mechanism for introducing full nodes, but we figured that it would cripple the node quite significantly (can't retrieve past logs and transactions) and also would hurt the network health, so until we figure out a way to keep the network guaranteed healthy we retained all history for now.]*

I think the reason behind 1024 previously was that it's low enough not to be a memory burden, and high enough to not be importable too fast. Now, with fast syncing where import times are an order of magnitude faster, maybe it would make sense to raise this. However a final value could be only tuned with a live network to see the latency effects. What value would you suggest for fast syncs? (Note, the name blockCacheLimit is a bit off, it would better be called resultCacheLimit as the downloader pulls not only blocks any more, but receipts too).